### PR TITLE
MultiSigDAOFactory and MultiSigDAO contracts with unit testing

### PR DESCRIPTION
**Pull request by shivamgarglab49**

### CODE REVIEW

File: contracts/dao/MultisigDAO.sol [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+contract MultiSigDAO is BaseDAO {
+    event TransactionCreated(bytes32 txnHash, TranscationInfo info);
+
+    enum TransactionState {
+        Pending,
+        Approved,
+        Executed
+    }
+
+    struct TranscationInfo {
```
Comment (amandeeprehal-lab49) [inline contracts/dao/MultisigDAO.sol:17]: Typo -  TranscationInfo -> TransactionInfo

File: contracts/gnosis/HederaGnosisSafe.sol [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                UINT_ZERO,
+                UINT_ZERO,
+                ADDRESS_ZERO,
+                ADDRESS_ZERO,
+                // Signature info
+                nonce
+            );
+            txHash = keccak256(txHashData);
+        }
+        require(checkSignatures(txHash), "Owner has not approved yet");
```
Comment (amandeeprehal-lab49) [inline contracts/gnosis/HederaGnosisSafe.sol:123]: pls add some comments here as well to explain why we extended function. 

The cost of on-chain approval is really cheap hence we there is no need of off-chain approval.  Also, Hashpack or equivalents wallets do not support ECDSA keys.

File: contracts/mock/MockBaseHTSWithTokenCreationFail.sol [hunk 3]
```diff
@@ -13,12 +13,12 @@ contract MockBaseHTSWithTokenCreationFail is IBaseHTS {
         address sender,
         address receiver,
         int256 amount
-    ) external override returns (int256 responseCode) {}
+    ) public override returns (int256 responseCode) {}
```
Comment (amandeeprehal-lab49) [inline contracts/mock/MockBaseHTSWithTokenCreationFail.sol:16]: we don't need these changes right?

**Review by amandeeprehal-lab49 — CHANGES_REQUESTED** [review]

File: contracts/common/BaseHTS.sol [hunk 4]
```diff
@@ -18,7 +18,7 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
         address sender,
         address receiver,
         int256 amount
-    ) external override returns (int256) {
+    ) public override returns (int256) {
```
[the remark this answers is not in the document]
Comment (shivamgarglab49) [reply contracts/common/BaseHTS.sol:21]: yes, inside HederaGnosisSafe.

File: contracts/gnosis/HederaGnosisSafe.sol [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        }
+    }
+
+    function transferTokenViaSafe(
+        address token,
+        address receiver,
+        uint256 amount
+    ) public returns (bool transferred) {
+        require(msg.sender == address(this), "GS031"); // only via safe txn
+        return super.transferToken(token, receiver, amount);
```
[the remark this answers is not in the document]
Comment (shivamgarglab49) [reply contracts/gnosis/HederaGnosisSafe.sol:54]: require already there.

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**